### DISCUSSION TITLE
Allow setting max_age/https_only of Session Middleware

### DISCUSSION THREAD

**Discussion by aersam** [zauberzeug/nicegui#1448] [Ideas / Feature Requests]
Would be nice if one could overwrite the SessionMiddleware's options in run_with Method. Eg, https_only, max_age etc.

Other option would be to allow disable the registration of Session Middleware, so that the user would have to do this on it's own

Could also do a PR

**Comment by aersam** [comment]
One can, of course do a hack:

```python

import nicegui.globals
from nicegui import ui

app = FastAPI()
ui.run_with(
    app,
    storage_secret=os.getenv("NICEGUI_STORAGE_SECRET", "fA3478GN$N$*wT60#--"),
)
nicegui.globals.app.user_middleware[0].options["max_age"] = 10
```

**Reply by bjlaur** [reply]
> ```python
> nicegui.globals.app.user_middleware[0].options["max_age"] = 10
> ```

Is there a 'hack' we can do just for app.storage.browser?

**Reply by rodja** [reply]
`app.storage.browser` is using the `SessionMiddleware` which is stored as `nicegui.globals.app.user_middleware[0]`. But we can not change it for the browser storage while leaving it as it is for other uses of `SessionMiddleware`.

**Comment by rodja** [comment]
Thanks for creating this feature request @aersam. It's a bit similar to https://github.com/zauberzeug/nicegui/discussions/991 but not identical. I'll cross-link both to make sure we think them together.

**Comment by rodja** [comment]
The feature may be related to #2837 which also needs a configurable time when storage data is removed.
